### PR TITLE
New zombie events

### PR DESCRIPTION
**Pull request by VALERA771**
Checklist:
* Test Consuming

### CODE REVIEW

**Comment by xNexus-ACS** [comment]
@VALERA771 the consuming event already exists

**Comment by xNexus-ACS** [comment]
and don't name Hurting to the Scp0492 Attack, bcs it will conflict with the normal one

**Comment by louis1706** [comment]
Scp0492 don't need this event Player HurtingEvent can cancel it too

**Comment by xNexus-ACS** [comment]
@VALERA771 lmao instead of putting the other event in Obsolete, just move it to the correct folder

**Comment by VALERA771** [comment]
Gonna test it one day

**Comment by xNexus-ACS** [comment]
if you can, please use the old transpiler, same for the old ConsumingEvent constructor @VALERA771

**Comment by VALERA771** [comment]
> if you can, please use the old transpiler, same for the old ConsumingEvent constructor @VALERA771 if you can, do not change anything in the consuming event

But why? And if remove = no changed, so then I can just close pr

**Comment by louis1706** [comment]
> > if you can, please use the old transpiler, same for the old ConsumingEvent constructor @VALERA771 if you can, do not change anything in the consuming event
> 
> But why? And if remove = no changed, so then I can just close pr

mostly do not make useless change

File: Exiled.Events/EventArgs/Scp0492/ConsumingCorpseEventArgs.cs [hunk 1]
```diff
@@ -26,11 +26,11 @@ public class ConsumingCorpseEventArgs : IPlayerEvent, IRagdollEvent, IDeniableEv
         /// <param name="error"> <inheritdoc cref="ErrorCode"/> </param>
         /// <param name="isAllowed"> <inheritdoc cref="IsAllowed"/> </param>
         /// <remarks> See <see cref="ZombieConsumeAbility.ConsumedRagdolls"/> for all RagDolls consumed. </remarks>
-        public ConsumingCorpseEventArgs(Player player, Ragdoll ragDoll, ZombieConsumeAbility.ConsumeError error, bool isAllowed = true)
+        public ConsumingCorpseEventArgs(ReferenceHub player, BasicRagdoll ragDoll, byte error, bool isAllowed = true)
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp0492/ConsumingCorpseEventArgs.cs:29]: keep ZombieConsumeAbility.ConsumeError and don't use byte

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by xNexus-ACS** [comment]
> > if you can, please use the old transpiler, same for the old ConsumingEvent constructor @VALERA771 if you can, do not change anything in the consuming event
> 
> But why? And if remove = no changed, so then I can just close pr

yeah ignore what I said, srry

File: Exiled.Events/Patches/Events/Scp0492/Consuming.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+                new(OpCodes.Callvirt, PropertyGetter(typeof(ConsumingCorpseEventArgs), nameof(ConsumingCorpseEventArgs.IsAllowed))),
+                new(OpCodes.Brtrue_S, retLabel),
+
+                new(OpCodes.Ret),
+
+                new CodeInstruction(OpCodes.Ldloc, ev.LocalIndex).WithLabels(retLabel),
+                new(OpCodes.Ldarg_0),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(ConsumingCorpseEventArgs), nameof(ConsumingCorpseEventArgs.ErrorCode))),
+                new(OpCodes.Stfld, Field(typeof(RagdollAbilityBase<ZombieRole>), nameof(RagdollAbilityBase<ZombieRole>._errorCode))),
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Scp0492/Consuming.cs:67]: Add comment to explain what the code does

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by joker-119** [comment]
> @VALERA771 lmao instead of putting the other event in Obsolete, just move it to the correct folder without creating a new one, same for the handlers and the patch

That creates a breaking change since the reference will change

**Comment by VALERA771** [comment]
> > @VALERA771 lmao instead of putting the other event in Obsolete, just move it to the correct folder without creating a new one, same for the handlers and the patch
> 
> That creates a breaking change since the reference will change

Any ways to remove breaking change or leave draft till new major release?

**Review by louis1706 — APPROVED** [review]
Accepted for Exiled7 (will just need a require testing)

**Comment by VALERA771** [comment]
7.0.0 soon. I'm making this not draft

**Comment by Misaka-ZeroTwo** [comment]
tested, works fine

**Review by louis1706 — APPROVED** [review]

**Comment by louis1706** [comment]
Transpiler need to be check again

**Comment by Misaka-ZeroTwo** [comment]
Do not merge due to break change

**Comment by VALERA771** [comment]
Yay! Milestone

**Pull request merged**